src/components/search/utils/useGetSearchResults.ts

```diff
@@ -1,12 +1,75 @@
-import { Result, useSearchState } from "@yext/search-headless-react";
+import { useState, useEffect } from "react";
+import {
+  Result,
+  SearchCore,
+  useSearchState,
+  useSearchActions,
+} from "@yext/search-headless-react";
+
+async function fetchAll<T>(core: SearchCore, verticalKey: string) {
+  async function fetchPage(offset: number) {
+    const res = await core.verticalSearch({
+      query: "",
+      verticalKey: verticalKey,
+      limit: 50,
+      offset,
+      retrieveFacets: false,
+      skipSpellCheck: true,
+    });
+    return res;
+  }
+
+  let offset = 0;
+  let done = false;
+  const out = [];
+
+  while (!done) {
+    const res = await fetchPage(offset);
```

Comment:
It takes a lot longer to fetch all results one page at a time. Can it fetch the first page and then all subsequent pages in parallel, like the Classic Pages locator did? https://gerrit.yext.com/plugins/gitiles/generator-ysp/+/refs/heads/master/components/Locator/src/js/Oracle.js#139
In that method, the parameter `data` is the result of the first fetch.

Reply:
Thanks, parallelized it.